File: src/com/lysdev/transperthcached/models/FavouriteStop.java
package com.lysdev.transperthcached.models;

import com.lysdev.transperthcached.activities.MainActivity;

public class FavouriteStop {
    private String stop_number;
    private Integer sid = null;
    private String description = null;

    public FavouriteStop(String stop_number) {
        this.stop_number = stop_number;
    }

    public FavouriteStop(Integer sid, String stop_number) {
        this(stop_number);
        this.sid = sid;
    }

    public String getStopNumber() {
        return this.stop_number;
    }

    public String getDescription() {
        if (this.description == null) {
            this.description = MainActivity.getConstantDB().getDescriptionOfStop(
                this.stop_number
            );
        }

        return this.description;
    }
}

File: src/com/lysdev/transperthcached/models/FavouriteStop.scala
package com.lysdev.transperthcached.models

import com.lysdev.transperthcached.activities.MainActivity

class FavouriteStop(val sid: Int, val stop_number: Int) {
    def this(stop_number: Integer) {
        this(0, stop_number)
    }

    lazy val description = (
        MainActivity.getConstantDB().getDescriptionOfStop(this.stop_number)
    )

    def getStopNumber = this.stop_number
    def getDescription = this.description

    // override def toString : String = String.format("%s - %s", this.stop_number)
}
